smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/api/request/service/SignUpForConsultingHourRequestServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package hu.unideb.smartcampus.service.api.request.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import hu.unideb.smartcampus.persistence.entity.ConsultingDateEntity;
+import hu.unideb.smartcampus.persistence.entity.UserConsultingDateEntity;
+import hu.unideb.smartcampus.persistence.entity.UserEntity;
+import hu.unideb.smartcampus.persistence.repository.ConsultingDateRepository;
+import hu.unideb.smartcampus.persistence.repository.UserConsultingDateRepository;
+import hu.unideb.smartcampus.persistence.repository.UserRepository;
+import hu.unideb.smartcampus.service.api.MessageProcessingClass;
+import hu.unideb.smartcampus.service.api.domain.response.wrapper.SignUpForConsultingHourWrapper;
+import hu.unideb.smartcampus.shared.requestmessages.BaseRequestType;
+import hu.unideb.smartcampus.shared.requestmessages.SignUpForConsultingHourRequest;
+import hu.unideb.smartcampus.shared.requestmessages.constants.RequestMessagesConstants;
+
+/**
+ * Processing signing up for consulting hours.
+ *
+ */
+@Service(SignUpForConsultingHourRequestServiceImpl.BEAN_NAME)
+@Transactional(propagation = Propagation.REQUIRED)
+public class SignUpForConsultingHourRequestServiceImpl
+    implements MessageProcessingClass<SignUpForConsultingHourWrapper> {
+
+  public static final String BEAN_NAME = "signUpForConsultingHourRequestServiceImpl";
+
+  private static final String USER_DOES_NOT_EXISTS = "User does not exists.";
+
+  private static final String NO_CONSULTING_DATE_EXISTS = "No consulting date exists.";
+
+  private static final String OK = "OK";
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(SignUpForConsultingHourRequestServiceImpl.class);
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @Autowired
+  private ConsultingDateRepository consultingDateRepository;
+
+  @Autowired
+  private UserConsultingDateRepository userConsultingDateRepository;
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public SignUpForConsultingHourWrapper getResponse(Object object) {
+    SignUpForConsultingHourRequest msg = (SignUpForConsultingHourRequest) object;
+    ConsultingDateEntity dateEntity = consultingDateRepository.findOne(msg.getConsultingHourId());
```

Comment:
Use services instead of repositories(Optional return value for the better null checking) 
findById and findByUsername